src/server/helpers/file.py

```diff
@@ -45,3 +46,10 @@ def write_file(
     path = fs.get_path(fullpath)
 
     return path
+
+
+def filter_file_path(path: str):
+    matches = parse_gitignore(".gitignore")  # type: ignore
```

Comment:
It will be parsing gitignore again and again for every file in the project. Also the path to gitignore is fs.get_fullpath('.gitignore')